### PR TITLE
CEP-221: Add createCollection if exists logic

### PR DESCRIPTION
**Pull request by dlazarenko-macrometa**
Added logic where it checks if collection exists before creating it.

### CODE REVIEW

File: src/main/java/com/c8db/internal/C8DatabaseImpl.java [hunk 1]
```diff
@@ -92,13 +92,20 @@ public C8Collection collection(final String name) {
...
     }
 
     @Override
     public CollectionEntity createCollection(final String name, final CollectionCreateOptions options)
             throws C8DBException {
-        return executor.execute(createCollectionRequest(name, options), CollectionEntity.class);
+        try {
+            return collection(name).getInfo();
+        } catch (final C8DBException e) {
+            if (C8Errors.ERROR_C8_DATA_SOURCE_NOT_FOUND.equals(e.getErrorNum())) {
```
Comment (grainier-macrometa) [inline src/main/java/com/c8db/internal/C8DatabaseImpl.java:104]: @dlazarenko-macrometa Why check for `ERROR_C8_DATA_SOURCE_NOT_FOUND`? Shouldn't it be `ERROR_C8_DATABASE_NOT_FOUND`?

**Review by grainier-macrometa — COMMENTED** [review]
@dlazarenko-macrometa Code change LGTM. However, should we really fix it here? Or since `C8Database.exists()` method is there, shouldn't it be fixed on the place we call `createCollection` (i.e. c8cep or c8streams)?

What if there's a use case, where a user only wants to create a new collection and needs an exception if that's already there?

cc: @StoyanTc @yehan-macrometa

File: src/main/java/com/c8db/internal/C8DatabaseImpl.java (hunk 1 above)
Comment (grainier-macrometa) [reply src/main/java/com/c8db/internal/C8DatabaseImpl.java:104]: Sorry, my bad it's `ERROR_C8_DATA_SOURCE_NOT_FOUND`.

**Comment by dlazarenko-macrometa** [comment]
> @dlazarenko-macrometa Code change LGTM. However, should we really fix it here? Or since `C8Collection.exists()` method is there, shouldn't it be fixed on the place we call `createCollection` (i.e. c8cep or c8streams)?
> 
> What if there's a use case, where a user only wants to create a new collection and needs an exception if that's already there?
> 
> cc: @StoyanTc @yehan-macrometa

Yep, I agree that it should be implemented in `c8db`, I am speaking with Sergey Georgiev to clarify with them.

**Comment by dlazarenko-macrometa** [comment]
Bug was moved to `c8db` team https://macrometa.atlassian.net/browse/DB-1218

**Pull request closed without merging**